src/core/modules/memory/memory_hooks.cpp

```diff
@@ -83,7 +85,7 @@ bool SP_HookHandler(HookType_t eHookType, CHook* pHook)
 	std::list<object> callbacks = g_mapCallbacks[pHook][eHookType];
 
 	// No need to do all this stuff, if there is no callback registered
-	if (callbacks.empty())
+	if (g_HooksDisabled || callbacks.empty())
```

Comment:
Not really dramatic but, since there is no need to query the map and initialize `callbacks` if the hooks are globally disabled, I think it would be better to move that check at the very top.

Reply:
Good point! I will do that later.